### DISCUSSION TITLE
.adf or .scp to .adf does not always result in a TOSEC match, is this normal?

### DISCUSSION THREAD

**Discussion by NebularNerd** [keirf/greaseweazle#167] [Q&A] [answered]
Hi everybody,

Pretty much as the title says, I've been testing the greaseweazel V4 (@keirf this thing is amazing) but before I archive all my Amiga disks I just wanted to see if this is normal. I've been making .scp files from my disks with the following settings;

`gw.exe read --retries=10 --revs=5 "N:\EMU\Floppy Archive\Amiga\diskname.scp::disktype=amiga"`

Reading around everyone seems to hint that revs=5 is required for 'archival' quality, disks are reading fine and aside from one slightly iffy read which passed when I re-read the disk there are no bad track issues, disks shows the nice PACMAN style layout in HxC's track viewer etc... 

However, most of the .scp's when converted to .adf are not being recognised by TOSEC in Romcenter, Workbench disks and Coverdiscs seem fine, but commercial games (originals that have always been write protected) are not. Is this an issue with how I'm reading them, or (and I think more likely) simply that the .adf's of the games are of better quality than the TOSEC database ones and I should just carry on and not worry about it. The .scp's work fine in WinUAE even heavily protected disks like Carrier Command run so I'm guessing all is well.

Has anybody else had similar experience?

**Comment by M1kerochip** [comment, marked as answer]
You can't convert commercial games to .adf.
The .adf format doesn't hold the copy protection information in the flux stream.

There are (more or less) no commercial .adf game disk images in TOSEC etc.
All the .adf files in TOSEC that are commercial games, have been cracked (had the copy protection removed) versions.

You'll find if you try to play those games, converted from .scp to .adf, they'll fail the copy protection checks, or crash etc.

You either need to use the .scp file in WinUAE etc, or convert them to hfe v3, to use with a gotek/hxc device.

TOSEC and no-intro, does though, have commercial game images, they're the SPS files.

Which disk images exactly are you comparing against?
The TOSEC files usually have the cracking group etc listed in the file name.

**Reply by NebularNerd** [reply]
Hi @M1kerochip 

I thought that was pretty much the case, I'll run up a couple of .ipf reads and see how they go, it's been a long time since I really looked at the whole .adf thing. From our other discussion it was a really long time ago which was why I was using .dms for trickier disks back in the day (which likely would have needing cracking anyway) , was not sure if the updated extended .adf format had support. 

I suppose this really then comes down to what format to store in, .ipf would get TOSEC love but it's essentially a closed format, .scp seems the better choice but you can't convert to .ipf (yet)


Edit: Doh! did not realise you cannot make .ipf's through greaseweazel (should have read the docs) and HxC does not let you export to .ipf either. I think I shall adopt the policy that if standard AmigaDOS discs are converting ok and matching TOSEC then my .scp files for all discs are very likely perfectly fine.

**Reply by M1kerochip** [reply]
The creator of the GW, also has a disk utils repository! 😎

One of the tools in there is disk-analyse that, for supported copy protection formats, let's you create (non official/sps) .ipf files!

No-intro are happy enough to start adding flux level game disk dumps. (I'll add some atari st ones this weekend I hope)

Its important to note that sps doesn't support all copy protection schemes either, and requires the sps library to work with the ipf files.
(it does support almost all the Amiga ones, but I think a few are still missing)

You can also still contact at least ome sps member, on eab and the Atari-forum, and submit Kryoflux dumps for conversion to official.ipf files.

The Greaseweazle supports dumping kf flux images too.

I'd ignore extended adf files, personally.

**Reply by NebularNerd** [reply]
I've read about the whole Kyroflux thing, I was looking to buy one for years but never got round to it. I've been reading the forums on EAB regarding the sps/ipf creation and it's a shame that it's been getting a bad rap about the image creation but now there seems to be only one active member working on it I can't see the situation improving. 

I think I'll stick with my previous approach, if I get a .scp with no read errors assume it's good, I might run up a copy of the disk-analyse to play with .ipf files but it's not really the end goal, more a would have been nice.

I'll also check out No-intro about the dumps to see if I can contribute.

Still think it's amazing how far we have come, I remember how excited I was when Toni of WinUAE managed to create his adfread utility back in the day after I suggested the idea (I was Oberth on EAB many years ago, lost the account and cannot reactivate) that we have come all the way from there to having a small, cheap PCB that can read everything is quite something.

**Reply by M1kerochip** [reply]
Hey @NebularNerd 
The kryofux tools let you check to see if a disk has been modified.

Ideally, obviously, if you're looking to preserve a game disk, a clean image with no errors is the first hurdle, and a disk that's not been written to is perfect for archival.

You can check the quality of the disk dump using HxC tools.

But, any disk image is better than no disk image!

It's also worth noting, too, that while an .ipf file is perfect for emulation etc, if you write an .ipf back to floppy it's got a marker, so it isn't 100% exactly the same as an original, and an SCP/KF etc flux stream should hopefully be.

But, yeah. 5 (at least) revolutions, no errors, no signs of having been written to flux image should be the ultimate goal.

I also wrote a small front end, for windows, to help me with dumping the disks, too :)

Oh, and some images I've made are up on Archive.org, too.

I put together a small script to generate, using HxC tools, a png of the disk surface, and data, so I can check each dump as I make them.

The front end just makes it easier to make multiple dumps, and automates putting each dump in it's own folder, and creating the .png files.
And set the SCP disk type bit.

And helps me keep track of what I've done!

I really should write a slightly more intuitive front end, but, well. It does the job I want it to!
Just like the Greaseweazle.

If you haven't noticed too, WinUAE added support for Robs floppy bridge, and it added support for the GW device.
So, you can, if you want to, now hook a real drive up, and read real disks, in WinUAE.

Which, for a lot of people, is the missing piece in emulation. The tactile feel of inserting a disk.

**Reply by keirf** [reply]
IPF is the gold standard for writing, really. It is purposely designed to be good for mastering to disk. All tracks have a write splice (mark) regardless of image source or write method.

**Reply by NebularNerd** [reply]
My long term goal is really just to preserve the disks at archival quality in the best format, if that is .ipf I can go back and re-run the ones I have already done. If they give similar file sizes to the .scp's I imagine they will still compress down fairly small with my crazy 7zip settings.

My main concern is longer term compatibility, outside of the sps/ipf library other .ipf support seems limited hence my going down the .scp route which seemed to offer much the same as .ipf but in a more open format.

@M1kerochip I did try your front end and could not get it to run gw, I'm currently using GreaseweazleGUI, but that appears to be discontinued. I'll take another look at yours and if I run into issues I'll chat to you on your side about that.

Most of my legit disks are in pretty much A+ condition data wise, aside from a couple of games that we had to save to such as Team Yankee as we did not have XCopy or DiskCopy at first to make backups, means they are obviously no longer original, the rest all show the typical signs of being an unmodified master. HxC track patterns do show some interesting stuff, it's interesting how most disks have the PACMAN style track layout but some that have been mastered by Guildhall Leisure for example have very different styles, legit unwritten on discs, but they do look more quite strange against the more common layout.

**Reply by keirf** [reply]
Best in my opinion for archive is a flux dump that has been validated using an analyser like disk-utilities for example.

**Reply by NebularNerd** [reply]
This is what I was doing, as the greaseweazel cannot create an .ipf from disk I'm using .scp files as the next best thing, WinUAE can use them as is even inside a .7z, protections seem to be intact even on games which I could never backup on an Amiga (Captive and Carrier Command for example are both pretty resistant to any X-Copy/D-Copy method) so for me that seems to be the way to go.

When dumped I check them in HxC's Track Viewer for bad sectors, when the odd one crops up I just give the disc another read, so far only one disk has 2 definite bad sectors but it passes a read without any retries (and it's nothing important enough on that disc to worry about)
